open-api.yaml

```diff
@@ -2227,6 +2263,142 @@ paths:
           description: Not Found
     parameters:
       - $ref: '#/components/parameters/indexUid'
+  '/indexes/{indexUid}/settings/pagination':
+    get:
+      operationId: indexes.settings.pagination.get
+      summary: Get pagination configuration
+      description: |
+        Get the pagination configuration of an index.
+      tags:
+        - Settings
+      security:
+        - apiKey: []
+      responses:
+        '200':
+          description: Ok
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/pagination'
+        '401':
+          $ref: '#/components/responses/401'
+        '404':
+          description: Not Found
+    patch:
+      operationId: indexes.settings.pagination.update
+      summary: Update pagination settings
+      description: |
+        Update the pagination configuration of an index.
+
+        > info
+        > If the provided index does not exist, it will be created.
+      tags:
+        - Settings
+      security:
+        - apiKey: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/pagination'
+            examples: {}
+        description: ''
+      responses:
+        '202':
+          $ref: '#/components/responses/202'
+        '401':
+          $ref: '#/components/responses/401'
+        '404':
+          description: Not Found
+      parameters:
+        - $ref: '#/components/parameters/Content-Type'
+    delete:
+      operationId: indexes.settings.pagination.reset
+      summary: Reset pagination settings to the default configuration
+      description: |
+        Reset the pagination settings of an index to its default configuration.
+      tags:
+        - Settings
+      security:
+        - apiKey: []
+      responses:
+        '202':
+          $ref: '#/components/responses/202'
+        '401':
+          $ref: '#/components/responses/401'
+        '404':
+          description: Not Found
+    parameters:
+      - $ref: '#/components/parameters/indexUid'
+  '/indexes/{indexUid}/settings/faceting':
+    get:
+      operationId: indexes.settings.faceting.get
+      summary: Get faceting configuration
+      description: |
+        Get the faceting configuration of an index.
+      tags:
+        - Settings
+      security:
+        - apiKey: []
+      responses:
+        '200':
+          description: Ok
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/faceting'
+        '401':
+          $ref: '#/components/responses/401'
+        '404':
+          description: Not Found
+    patch:
```

Comment:
Following the [spec #152](https://github.com/meilisearch/specifications/pull/152) we should use `PUT` for these routes, correct?

Reply:
I had a doubt about using PATCH/PUT for the following reason.

There is a very high chance that we will add parameters to the `pagination` and `facetting` objects in the future, by setting PATCH I wanted to avoid a future breaking and thus directly allow to edit each of the properties partially.

What do you think Bruno? I have alternated between PUT and PATCH to choose PATCH.

Reply:
I understand your point now, and it seems a good choice from my side. Making decisions with data that does not exist (yet) is hard because you never know when you are doing overengineering. 
I don't know if these new attributes will appear in the future but I trust you! 🤝